### PR TITLE
Engine block ordering

### PR DESCRIPTION
**Pull request by keorn**
New method on `Engine` that lets `Blockchain` know about the ordering. #3489

### CODE REVIEW

File: ethcore/src/ethereum/ethash.rs [hunk 1]
```diff
@@ -325,6 +327,15 @@ impl Engine for Ethash {
 	fn verify_transaction(&self, t: &SignedTransaction, _header: &Header) -> Result<(), Error> {
 		t.sender().map(|_|()) // Perform EC recovery and cache sender
 	}
+
+	fn is_new_best_block(&self, best_total_difficulty: U256, _best_header: HeaderView, parent_details: &BlockDetails, new_header: &HeaderView) -> bool {
+		is_new_best_block(best_total_difficulty, parent_details, new_header)
```
Comment (keorn) [inline ethcore/src/ethereum/ethash.rs]: The same as default, but kept to be explicit.

**Comment by rphmeier** [comment]
nice! I wonder if in the future it'll be feasible to extend this abstraction even to stuff like difficulty which is only relevant to PoW.

**Comment by keorn** [comment]
Yep, that would be nice (and will probably need to be done). Currently all additional info has to be stuffed into the seal.

**Comment by gavofyork** [comment]
this still compiles with IPC enabled, right?

**Comment by keorn** [comment]
Ah, you are right `snapshot_service_trait` does not compile. Not too familiar with the IPC setup.

**Comment by keorn** [comment]
Actually it was broken before this PR.

**Comment by rphmeier** [comment]
I broke it earlier with #2935 and also the `list_storage` stuff broke it as well

Preparing workarounds. The codegen implementation could really use some love...

**Pull request merged**